app/Post_Types/Archive_Item/Rest.php

```diff
@@ -21,21 +21,103 @@ class Rest {
 	const FIELD_INDEX_TYPE = 'type';
 	const FIELD_INDEX_FIELD_ID = 'field_id';
 	const FIELD_INDEX_FIELD_POPUP = 'field_popup';
+	const FIELD_INDEX_FIELD_ELASTIC = 'field_elastic';
 
 	protected $container;
 	protected $fields;
 
-	public function init() {
-		$this->fields = $this->get_fields();
-	}
-
 	public function hooks() {
 		$achive_item_name = Archive_Item::NAME;
-		add_filter( "rest_{$achive_item_name}_query", [$this, 'rest_api_filter_add_vars'], 10, 2 );
-		add_filter( "rest_{$achive_item_name}_query", [$this, 'rest_api_filter_add_order_by'], 3, 2 );
+		add_filter( "rest_{$achive_item_name}_query", [ $this, 'rest_api_filter_add_vars' ], 10, 2 );
+		add_filter( "rest_{$achive_item_name}_query", [ $this, 'rest_api_filter_add_order_by' ], 3, 2 );
+		add_filter( "rest_{$achive_item_name}_query", [ $this, 'rest_api_enable_elastic' ], 11, 2 );
+		add_filter( 'ep_formatted_args_query', [ $this, 'ep_formatted_args_query' ], 10, 2 );
+		add_filter( 'ep_post_sync_args', [ &$this, 'custom_ep_post_sync_args' ], 10, 2);
```

Comment:
There is no reason to use `&$this` here or anywhere: objects are always and only passed by reference, not copy.